src/constitutive/TACSSmearedCompositeShellConstitutive.h

```diff
@@ -0,0 +1,145 @@
+  Additional copyright (C) 2010 Graeme J. Kennedy and Joaquim
+  R.R.A. Martins All rights reserved.
+
+  TACS is licensed under the Apache License, Version 2.0 (the
+  "License"); you may not use this software except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+#ifndef TACS_SMEARED_COMPOSITE_SHELL_CONSTITUTIVE_H
+#define TACS_SMEARED_COMPOSITE_SHELL_CONSTITUTIVE_H
+
+#include "TACSMaterialProperties.h"
+#include "TACSShellConstitutive.h"
+
+class TACSSmearedCompositeShellConstitutive : public TACSShellConstitutive {
+ public:
+  TACSSmearedCompositeShellConstitutive(
+      int _num_plies, TACSOrthotropicPly **_ply_props, TacsScalar _thickness,
+      const TacsScalar *_ply_angles, const TacsScalar *_ply_fractions,
+      int _thickness_dv_num = -1, const int *_ply_fraction_dv_nums = NULL,
+      TacsScalar _thickness_lb = 0.0, TacsScalar _thickness_ub = 1e20,
+      const TacsScalar *_ply_fraction_lb = NULL,
+      const TacsScalar *_ply_fraction_ub = NULL);
+  ~TACSSmearedCompositeShellConstitutive();
+
+  // Retrieve the global design variable numbers
+  int getDesignVarNums(int elemIndex, int dvLen, int dvNums[]);
+
+  // Set the element design variable from the design vector
+  int setDesignVars(int elemIndex, int dvLen, const TacsScalar dvs[]);
+
+  // Get the element design variables values
+  int getDesignVars(int elemIndex, int dvLen, TacsScalar dvs[]);
+
+  // Get the lower and upper bounds for the design variable values
+  int getDesignVarRange(int elemIndex, int dvLen, TacsScalar lb[],
+                        TacsScalar ub[]);
+
+  // Retrieve the design variable for plotting purposes
+  TacsScalar evalDesignFieldValue(int elemIndex, const double pt[],
+                                  const TacsScalar X[], int index);
+
+  // Evaluate the material density
+  TacsScalar evalDensity(int elemIndex, const double pt[],
+                         const TacsScalar X[]);
+
+  // Add the derivative of the density
+  void addDensityDVSens(int elemIndex, TacsScalar scale, const double pt[],
+                        const TacsScalar X[], int dvLen, TacsScalar dfdx[]);
+
+  // Evaluate the mass moments
+  void evalMassMoments(int elemIndex, const double pt[], const TacsScalar X[],
+                       TacsScalar moments[]);
+
+  // Add the sensitivity of the mass moments
+  void addMassMomentsDVSens(int elemIndex, const double pt[],
+                            const TacsScalar X[], const TacsScalar scale[],
+                            int dvLen, TacsScalar dfdx[]);
+
+  // Evaluate the specific heat
+  TacsScalar evalSpecificHeat(int elemIndex, const double pt[],
+                              const TacsScalar X[]);
+
+  // Evaluate the stress
+  void evalStress(int elemIndex, const double pt[], const TacsScalar X[],
+                  const TacsScalar strain[], TacsScalar stress[]);
+
+  // Add the contribution
```

Comment:
Incomplete docstring

Reply:
@A-CGray, could you clarify this a bit? What are you expecting here?

Reply:
Nvm, I assume you mean adding docstring parameters of the C++ side? 
I can do this, but I'm noticing there's a number of classes that are missing these docstrings as well ([TACSIsoShellConstitutive.h](https://github.com/smdogroup/tacs/blob/master/src/constitutive/TACSIsoShellConstitutive.h), [TACSIsoTubeBeamConstitutive.h](https://github.com/smdogroup/tacs/blob/master/src/constitutive/TACSIsoTubeBeamConstitutive.h), etc).

We should go back (in a separate PR) and make this standard consistent across all C++ classes

Reply:
I just meant that this docstring ("Add the contribution") doesn't fully describe what the function does. You're right that we should probably go through and standardize how we write docstrings in the header and source files respectively though

Reply:
I fixed the truncated docstring here. We can leave documenting the parameters consistently across all C++ classes for another PR